### DISCUSSION TITLE
New fly system

### DISCUSSION THREAD

**Discussion by Ronal1999** [Zrips/CMI#7458] [Feature Requests]
Hello, I was thinking if you can change the /fly system, but now it is for how many blocks you are away from others to give an example.

If a user has /fly but has a restriction of 150 blocks in X and Z, this restriction refers to the fact that if a user near you at 150 blocks in X and Z will deactivate said /fly and you will fall if you are flying or not. activate the flight mode until you move away from said users close to you, of course, if for example you are at a height of 100 and someone approaches and the flight mode is deactivated you will not die by falling that time. Also, if they can be added for permissions, I will leave the confi.yml as it would be in case this request is accepted.

enable-fly: '&efly has been &aenable'
disable-fly: '&efly has been &cdisable'

#message that will pop out every time there is somebody near you
move-away: '&eYou have to move %distance% blocks from other users'
list:
#this is a list of users that have /fly enable or disable, this will help staff from networks to know who is using the command

'&eUsers with /fly'
element: '&e%status% %user_name%: %radio%'
user:
#how long is gonna be detected when users are near you
fly-detect-seconds: 1
flys-peed: 1
#numbers in X and Y always have to be the same
blocks:
x:50
y:300
z:50
#so permission for using this will be cmi.command.fly.user

vip:
#how long is gonna be detected when users are near you
fly-detect-seconds: 3
flys-peed: 1
#numbers in X and Z always have to be the same
blocks:
x:100
y:300
z:100
#so permission for using this will be cmi.command.fly.vip

so if you don't wanna give any restrictions with /fly just permission cmi.command.fly

and the last thing add support for UClans, Guilds, and others pls thanks, so /fly command won't be deactivated when a player near you, is from your clan

Thank you very much I know I am asking a lot but this will make server more dynamic

**Comment by MrCrazys** [comment]
In my opinion, this seems a bit out of the scope of CMI. Probably best just to have a plugin custom coded for your needs.
